### PR TITLE
Add gambit-scheme

### PR DESCRIPTION
**Pull request by ober**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/gambit-scheme.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    args = %W[
+      --prefix=#{prefix}
+      --enable-single-host
+      --enable-multiple-versions
+      --enable-default-runtime-options=f8,-8,t8
+      --enable-poll
+      --enable-openssl
+    ]
+
+    inreplace "lib/os_io.c" do |s|
```
Comment (fxcoudert) [inline Formula/gambit-scheme.rb]: This bug needs to be reported and fixed upstream.

File: Formula/gambit-scheme.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  sha256 "667ae2ee657c22621a60b3eda6e242224d41853adb841e6ff9bc779f19921c18"
+
+  depends_on "openssl"
+
+  def install
+    args = %W[
+      --prefix=#{prefix}
+      --enable-single-host
+      --enable-multiple-versions
+      --enable-default-runtime-options=f8,-8,t8
```
Comment (fxcoudert) [inline Formula/gambit-scheme.rb]: Why are we not building with default options?

File: Formula/gambit-scheme.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    ]
+
+    inreplace "lib/os_io.c" do |s|
+      s.gsub! "SSL_CTX_set_default_verify_paths (c->tls_ctx);", ""
+      s.gsub! "SSL_CTX_set_verify (c->tls_ctx, SSL_VERIFY_PEER, NULL);", ""
+    end
+
+    system "./configure", *args
+    system "make"
+    ENV.deparallelize
```
Comment (fxcoudert) [inline Formula/gambit-scheme.rb]: Failure to run make in parallel are bugs, and should be reported upstream.

File: Formula/gambit-scheme.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    system "make", "install"
+  end
+
+  test do
+    assert_equal "0123456789", shell_output("#{bin}/gsi -e '(for-each write '(0 1 2 3 4 5 6 7 8 9))'")
+    assert_equal 0, $CHILD_STATUS.exitstatus
```
Comment (fxcoudert) [inline Formula/gambit-scheme.rb]: Good test!
`shell_output` already checks the exit status, this line can be removed.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Comment by fxcoudert** [comment]
```
==> /usr/local/Cellar/gambit-scheme/4.9.1/bin/gsi -e '(for-each write '(0 1 2 3 4 5 6 7 8 9))'
sh: -c: line 0: syntax error near unexpected token `('
sh: -c: line 0: `/usr/local/Cellar/gambit-scheme/4.9.1/bin/gsi -e '(for-each write '(0 1 2 3 4 5 6 7 8 9))''
Error: gambit-scheme: failed
```

File: Formula/gambit-scheme.rb (hunk 3 above)
Comment (ober) [reply Formula/gambit-scheme.rb]: Done.

File: Formula/gambit-scheme.rb (hunk 2 above)
Comment (ober) [reply Formula/gambit-scheme.rb]: This is the best use case for a typical user.

File: Formula/gambit-scheme.rb (hunk 1 above)
Comment (ober) [reply Formula/gambit-scheme.rb]: Will remove this.

File: Formula/gambit-scheme.rb (hunk 2 above)
Comment (fxcoudert) [reply Formula/gambit-scheme.rb]: Is this the setting that upstream ships by default?

**Pull request closed without merging**